engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateByWindowTime.java

```diff
@@ -261,20 +264,22 @@ public void computeAffectedRowsAndOperators(UpdateByWindowBucketContext context,
         final WritableRowSet tmpAffected = RowSetFactory.empty();
 
         if (upstream.modified().isNonempty()) {
-            // modified timestamps will affect the current and previous values
             try (final RowSet modifiedAffected =
                     computeAffectedRowsTime(ctx, upstream.modified(), prevUnits, fwdUnits, false)) {
                 tmpAffected.insert(modifiedAffected);
             }
-            try (final WritableRowSet modifiedAffectedPrev =
-                    computeAffectedRowsTime(ctx, upstream.getModifiedPreShift(), prevUnits, fwdUnits, true)) {
-                // we used the SSA (post-shift) to get these keys, no need to shift
-                // retain only the rows that still exist in the sourceRowSet
-                modifiedAffectedPrev.retain(ctx.timestampValidRowSet);
-                tmpAffected.insert(modifiedAffectedPrev);
+            if (ctx.timestampsModified) {
```

Comment:
Don't we have to be concerned with mods to input columns for the operators, too?